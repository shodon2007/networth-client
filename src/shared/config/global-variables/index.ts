import globalEnv from "./global-variables";

export default globalEnv;
