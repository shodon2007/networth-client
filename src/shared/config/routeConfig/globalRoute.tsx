import {Layout} from "src/shared/ui/Layout/Layout";
import routeConfig, {RouteItem} from "./routeConfig";


const globalRoute: RouteItem = {
	path: routeConfig.routePath.home,
	element: <Layout />,
	index: true,
};

export default globalRoute;
