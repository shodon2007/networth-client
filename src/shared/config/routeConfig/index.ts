export {AppRoutes, routePath, routes} from "./routeConfig";
export type {RouteItem} from "./routeConfig";
