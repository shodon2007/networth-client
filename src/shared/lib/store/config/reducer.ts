import {combineReducers} from "@reduxjs/toolkit";
import {themeReducer} from "src/app/providers/themeProvider";
import userSlice from "src/entities/user/services/userSlice";

export const rootReducer = combineReducers({
	theme: themeReducer,
	user: userSlice,
});
