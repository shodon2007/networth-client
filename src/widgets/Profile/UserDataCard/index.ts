import { UserDataCard } from "./ui/UserDataCard";

export { UserDataCard }