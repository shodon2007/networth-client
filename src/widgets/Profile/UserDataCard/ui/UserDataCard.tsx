import {useState} from "react";
import {useTranslation} from "react-i18next";
import {useUser} from "src/entities/user";
import DetailIcon from "src/shared/assets/DetailIcon.svg";
import Mailicon from "src/shared/assets/MailIcon.svg";
import Block from "src/shared/ui/Block/Block";
import Button, {ThemeButton} from "src/shared/ui/Button/Button";

import ChangeAvatarModal from "./ChangeAvatarModal";
import ChangeEmailModal from "./ChangeEmailModal";
import ChangePasswordModal from "./ChangePasswordModal";
import EditUserModal from "./EditUserModal";
import cls from "./UserCardStyle.module.scss";

export const UserDataCard = () => {
	const {data: userData, isAuth} = useUser();
	const [editModalOpen, setEditModalOpen] = useState(false);
	const [changeAvatarModalOpen, setChangeAvatarModalOpen] = useState(false);
	const [changePasswordModalOpen, setChangePasswordModalOpen] = useState(false);
	const [changeEmailModalOpen, setChangeEmailModalOpen] = useState(false);
	const {t} = useTranslation();

	if (!userData || !isAuth) {
		return <div>Пожалуйста Зарегестрируйтесь</div>;
	}

	return (
		<Block className={cls.UserCard}>
			<div className={cls.UserCard__info}>
				<div className={cls.avatar}>
					<img src={userData.avatar} alt="Avatar" className={cls.avatar__img} />
					<div className={cls.avatar__menu}>
						<Button onClick={() => setChangeAvatarModalOpen(true)}>
							{t("profile.updateAvatar")}
						</Button>
					</div>
				</div>

				<ul className={cls.UserCard__userInfo}>
					<li className={cls.UserCard__name}>
						{`${userData.name} ${userData.surname}`}
					</li>
					<li className={cls.UserCard__detail}>
						<img src={Mailicon} />
						{userData?.email}
					</li>
					<li className={cls.UserCard__detail}>
						<img src={DetailIcon} alt="" />
						<h4>Detils</h4>
					</li>
				</ul>
			</div>
			<div className={cls.UserCard__buttons}>
				<Button
					theme={ThemeButton.SMALL}
					onClick={() => setEditModalOpen(true)}
				>
					{t("profile.editProfile")}
				</Button>
				<Button
					theme={ThemeButton.SMALL}
					onClick={() => setChangePasswordModalOpen(true)}
				>
					{t("profile.editPassword")}
				</Button>
				<Button
					theme={ThemeButton.SMALL}
					onClick={() => setChangeEmailModalOpen(true)}
				>
					{t("profile.editEmail")}
				</Button>
			</div>
			<EditUserModal
				isOpen={editModalOpen}
				close={() => setEditModalOpen(false)}
			/>
			<ChangeAvatarModal
				isOpen={changeAvatarModalOpen}
				close={() => setChangeAvatarModalOpen(false)}
			/>
			<ChangePasswordModal
				isOpen={changePasswordModalOpen}
				close={() => setChangePasswordModalOpen(false)}
			/>
			<ChangeEmailModal
				isOpen={changeEmailModalOpen}
				close={() => setChangeEmailModalOpen(false)}
			/>
		</Block>
	);
};
