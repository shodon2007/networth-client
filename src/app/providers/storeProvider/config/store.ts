import {configureStore} from "@reduxjs/toolkit";
import {rootReducer} from "src/shared/lib/store/config/reducer";
import {userApi} from "src/entities";
import {setupListeners} from "@reduxjs/toolkit/query";
import {profileApi} from "src/entities/Profile";

const store = configureStore({
	reducer: rootReducer,
	middleware: (getDefaultMiddleware) => {
		return getDefaultMiddleware()
			.concat(userApi.middleware)
			.concat(profileApi.middleware);
	},
});
setupListeners(store.dispatch);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
