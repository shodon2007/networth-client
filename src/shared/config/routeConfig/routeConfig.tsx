import {ReactNode} from "react";
import {ActivationPageAsync} from "src/pages/Auth/Activation";
import {LoginPageAsync} from "src/pages/Auth/Login";
import {RegistrationPageAsync} from "src/pages/Auth/Registration";
import {FeedPageAsync} from "src/pages/Feed";
import {MessengerPageAsync} from "src/pages/Messenger";
import {ProfilePageAsync} from "src/pages/Profile";
import {Layout} from "src/shared/ui/Layout/Layout";

export enum AppRoutes {
	REGISTRATION = "registration",
	LOGIN = "login",
	ACTIVATION = "activation",

	HOME = "home",
	PROFILE = "profile",
	MESSENGER = "messenger",
}

export const routePath: Record<AppRoutes, string> = {
	[AppRoutes.HOME]: "",
	[AppRoutes.PROFILE]: "/profile",
	[AppRoutes.LOGIN]: "/login",
	[AppRoutes.REGISTRATION]: "/registration",
	[AppRoutes.ACTIVATION]: "/activation",
	[AppRoutes.MESSENGER]: "/messanger",
};

export interface RouteItem {
	path: string;
	element: ReactNode;
	index?: boolean;
	children?: RouteItem[];
}

export const routes: RouteItem[] = [
	{
		path: routePath.home,
		element: <Layout />,
		children: [
			{
				path: routePath.home,
				element: <FeedPageAsync />,
				index: true,
			},
			{
				path: routePath.profile,
				element: <ProfilePageAsync />,
				index: true,
			},
			{
				path: routePath.messenger,
				element: <MessengerPageAsync />,
				index: true,
			},
			{
				path: routePath.activation,
				element: <ActivationPageAsync />,
			},
			{
				path: routePath.registration,
				element: <RegistrationPageAsync />,
			},
			{
				path: routePath.login,
				element: <LoginPageAsync />,
			},
		],
	},
];
