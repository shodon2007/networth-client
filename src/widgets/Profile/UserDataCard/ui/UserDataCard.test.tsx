import {renderTestApp} from "src/shared/tests";
import {UserDataCard} from "./UserDataCard";
import {routePath} from "src/shared/config/routeConfig";
import {waitFor} from "@testing-library/dom";
import userEvent from "@testing-library/user-event";

const initialStore = {
	user: {
		isAuth: true,
		accessToken: "hahahah",
		refreshToken: "hehehehe",
		data: {
			email: "shodon",
			id: 1,
			isActivated: 1,
			name: "shodon",
			phoneNumber: "fdsfdf",
			surname: "asrorov",
		},
		isUserDataLoading: false,
	},
};

describe("testing UserDataCard", () => {
	it("should match the snapshot", async () => {
		const screen = renderTestApp(routePath.profile, initialStore);
		expect(await screen.findByTestId("user-data-card")).toBeInTheDocument();
		expect(screen.getByTestId("user-data-card")).toMatchSnapshot();
	});
	it("should show user-data correct", async () => {
		const screen = renderTestApp(routePath.profile, initialStore);
		expect(await screen.findByTestId("profile-fullname")).toContainHTML(
			`${initialStore.user.data.name} ${initialStore.user.data.surname}`,
		);
		expect(await screen.findByTestId("profile-email")).toContainHTML(
			initialStore.user.data.email,
		);
	});
	it("should be edit profile modal works", async () => {
		const screen = renderTestApp(routePath.profile, initialStore);
		let editProfileButton: HTMLElement;
		await waitFor(
			() => (editProfileButton = screen.getByTestId("editProfileButton")),
		);
		expect(editProfileButton).toBeInTheDocument();
		expect(screen.queryByTestId("editUserModal")).not.toBeInTheDocument();
		const user = userEvent.setup();
		await user.click(editProfileButton);
		expect(screen.getByTestId("editUserModal")).toBeInTheDocument();
		const cancelButton = screen.getByTestId("profile-cancel-button");
		await user.click(cancelButton);
		expect(screen.queryByTestId("editUserModal")).not.toBeInTheDocument();
	});

	it("should be edit changePassword modal works", async () => {
		const screen = renderTestApp(routePath.profile, initialStore);
		let changePasswordButton: HTMLElement;
		await waitFor(
			() => (changePasswordButton = screen.getByTestId("changePasswordButton")),
		);
		expect(changePasswordButton).toBeInTheDocument();
		expect(screen.queryByTestId("changePasswordModal")).not.toBeInTheDocument();
		const user = userEvent.setup();
		await user.click(changePasswordButton);
		expect(screen.getByTestId("changePasswordModal")).toBeInTheDocument();
		const cancelButton = screen.getByTestId("profile-cancel-button");
		await user.click(cancelButton);
		expect(screen.queryByTestId("changePasswordModal")).not.toBeInTheDocument();
	});

	it("should be edit changeEmail modal works", async () => {
		const screen = renderTestApp(routePath.profile, initialStore);
		let changePasswordButton: HTMLElement;
		await waitFor(
			() => (changePasswordButton = screen.getByTestId("changeEmailButton")),
		);
		expect(changePasswordButton).toBeInTheDocument();
		expect(screen.queryByTestId("changeEmailModal")).not.toBeInTheDocument();
		const user = userEvent.setup();
		await user.click(changePasswordButton);
		expect(screen.getByTestId("changeEmailModal")).toBeInTheDocument();
		const cancelButton = screen.getByTestId("profile-cancel-button");
		await user.click(cancelButton);
		expect(screen.queryByTestId("changeEmailModal")).not.toBeInTheDocument();
	});
});
