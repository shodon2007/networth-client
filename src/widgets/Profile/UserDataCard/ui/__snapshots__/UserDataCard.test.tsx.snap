// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserDataCard should match the snapshot 1`] = `
<div
  data-testid="user-data-card"
>
  <div
    class="block default userCard"
    data-testid="block"
  >
    <div
      class="userCardInfo"
    >
      <div
        class="avatar"
      >
        <img
          alt="Avatar"
          class="avatarImg"
        />
        <div
          class="avatarMenu"
        >
          <button
            class="button"
          >
            Обновить фотографию
          </button>
        </div>
      </div>
      <ul
        class="userCardUserInfo"
      >
        <li
          class="userCardName"
          data-testid="profile-fullname"
        >
          shodon asrorov
        </li>
        <li
          class="userCardDetail"
          data-testid="profile-email"
        >
          <img
            src=""
          />
          shodon
        </li>
        <li
          class="userCardDetail"
        >
          <img
            alt=""
            src=""
          />
          <h4>
            Detils
          </h4>
        </li>
      </ul>
    </div>
    <div
      class="userCardButtons"
    >
      <button
        class="button small"
        data-testid="editProfileButton"
      >
        Изменить профиль
      </button>
      <button
        class="button small"
        data-testid="changePasswordButton"
      >
        Изменить пароль
      </button>
      <button
        class="button small"
        data-testid="changeEmailButton"
      >
        Изменить почту
      </button>
    </div>
  </div>
</div>
`;
